### PR TITLE
Simplified and extended jobs tests

### LINKED ISSUE #8584: Test flake in jobs extended test
Extended test uses the DockerHub perl image, which takes more than 2minutes to pull sometimes, which causes it to fail.

We should switch to an image we have and do a simpler test (centos or origin, test bash).

@soltysh can you knock this one out real quick so we can finally get origin_extended passing?

```
[job] openshift can execute jobs controller 
  should create and run a job in user project
  /data/src/github.com/openshift/origin/test/extended/jobs/jobs.go:47

STEP: Creating a kubernetes client
Apr 20 20:10:40.022: INFO: >>> testContext.KubeConfig: /tmp/openshift/test-extended/core/openshift.local.config/master/admin.kubeconfig

STEP: Building a namespace api object
Apr 20 20:10:40.046: INFO: configPath is now "/tmp/extended-test-job-controller-uppzo-egdz2-user.kubeconfig"
Apr 20 20:10:40.046: INFO: The user is now "extended-test-job-controller-uppzo-egdz2-user"
Apr 20 20:10:40.046: INFO: Creating project "extended-test-job-controller-uppzo-egdz2"
STEP: Waiting for a default service account to be provisioned in namespace
Apr 20 20:10:40.397: INFO: Waiting up to 2m0s for service account default to be provisioned in ns extended-test-job-controller-uppzo-egdz2
Apr 20 20:10:40.398: INFO: Service account default in ns extended-test-job-controller-uppzo-egdz2 with secrets found. (1.100568ms)
STEP: creating a job from "/data/src/github.com/openshift/origin/test/extended/fixtures/job-controller.yaml"
Apr 20 20:10:40.398: INFO: Running 'oc create --namespace=extended-test-job-controller-uppzo-egdz2 --config=/tmp/extended-test-job-controller-uppzo-egdz2-user.kubeconfig -f /data/src/github.com/openshift/origin/test/extended/fixtures/job-controller.yaml'
job "pi" created
STEP: Waiting for pod...
STEP: Collecting events from namespace "extended-test-job-controller-uppzo-egdz2".
Apr 20 20:12:40.489: INFO: At 2016-04-20 20:10:40 -0400 EDT - event for pi: {job-controller } SuccessfulCreate: Created pod: pi-4oaao
Apr 20 20:12:40.489: INFO: At 2016-04-20 20:10:40 -0400 EDT - event for pi-4oaao: {default-scheduler } Scheduled: Successfully assigned pi-4oaao to 172.18.14.36
Apr 20 20:12:40.489: INFO: At 2016-04-20 20:12:16 -0400 EDT - event for pi-4oaao: {kubelet 172.18.14.36} Pulling: pulling image "perl"
Apr 20 20:12:40.494: INFO: POD                         NODE          PHASE    GRACE  CONDITIONS
Apr 20 20:12:40.494: INFO: docker-registry-1-cdws8     172.18.14.36  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-04-20 19:56:28 -0400 EDT  }]
Apr 20 20:12:40.494: INFO: router-1-deploy             172.18.14.36  Failed          []
Apr 20 20:12:40.494: INFO: router-2-zd6xz              172.18.14.36  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-04-20 19:58:26 -0400 EDT  }]
Apr 20 20:12:40.494: INFO: update-demo-nautilus-ua2dc  172.18.14.36  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-04-20 20:12:39 -0400 EDT ContainersNotReady containers with unready status: [update-demo]}]
Apr 20 20:12:40.494: INFO: update-demo-nautilus-yec2m  172.18.14.36  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-04-20 20:12:32 -0400 EDT  }]
Apr 20 20:12:40.494: INFO: cephfs-server               172.18.14.36  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-04-20 20:12:14 -0400 EDT ContainersNotReady containers with unready status: [cephfs-server]}]
Apr 20 20:12:40.494: INFO: pi-4oaao                    172.18.14.36  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-04-20 20:12:14 -0400 EDT ContainersNotReady containers with unready status: [pi]}]
Apr 20 20:12:40.494: INFO: 
Apr 20 20:12:40.497: INFO: 
Logging node info for node 172.18.14.36
Apr 20 20:12:40.500: INFO: Node Info: &{{ } {172.18.14.36   /api/v1/nodes/172.18.14.36 52e593fc-0753-11e6-9479-0eec20cd1421 8793 0 2016-04-20 19:55:13 -0400 EDT <nil> <nil> map[kubernetes.io/hostname:172.18.14.36] map[]} { 172.18.14.36  false} {map[cpu:{2.000 DecimalSI} memory:{3778179072.000 BinarySI} pods:{110.000 DecimalSI}] map[pods:{110.000 DecimalSI} cpu:{2.000 DecimalSI} memory:{3778179072.000 BinarySI}]  [{OutOfDisk False 2016-04-20 20:12:31 -0400 EDT 2016-04-20 19:55:13 -0400 EDT KubeletHasSufficientDisk kubelet has sufficient disk space available} {Ready True 2016-04-20 20:12:31 -0400 EDT 2016-04-20 19:55:23 -0400 EDT KubeletReady kubelet is posting ready status}] [{LegacyHostIP 172.18.14.36} {InternalIP 172.18.14.36}] {{10250}} {f9370ed252a14f73b014c1301a9b6d1b EC259C34-B7F4-EB54-A98C-027F9079DD69 18b06665-6b3a-4bc0-ad32-51b18eb95193 3.10.0-229.7.2.el7.x86_64 Red Hat Enterprise Linux Server 7.2 (Maipo) docker://1.9.1 v1.2.0-36-g4a3f9c5 v1.2.0-36-g4a3f9c5} [{[<none>:<none>] 443321353} {[<none>:<none>] 543758488} {[172.30.110.46:5000/extended-test-forcepull-h7l9h-eqxgy/forcepull-extended-test-builder:latest] 443310209} {[openshift/deployment-example:v2] 7585584} {[openshift/deployment-example:v1] 7585584} {[openshift/hello-openshift:de5b801 openshift/hello-openshift:latest] 7584592} {[docker.io/openshift/hello-openshift:latest] 7584592} {[openshift/origin-custom-docker-builder:de5b801 openshift/origin-custom-docker-builder:latest] 402543638} {[openshift/node:de5b801 openshift/node:latest] 636131342} {[openshift/origin-f5-router:de5b801 openshift/origin-f5-router:latest] 543747344} {[openshift/origin-sti-builder:de5b801 openshift/origin-sti-builder:latest] 543747344} {[openshift/origin-gitserver:latest openshift/origin-gitserver:de5b801] 581450216} {[openshift/origin-docker-builder:de5b801 openshift/origin-docker-builder:latest] 543747344} {[openshift/origin-recycler:de5b801 openshift/origin-recycler:latest] 34975664} {[docker.io/openshift/origin-deployer:latest] 543747344} {[openshift/origin-deployer:de5b801 openshift/origin-deployer:latest] 543747344} {[openshift/origin-egress-router:de5b801 openshift/origin-egress-router:latest] 298250070} {[openshift/origin-docker-registry:de5b801 openshift/origin-docker-registry:latest] 341697712} {[openshift/origin-keepalived-ipfailover:de5b801 openshift/origin-keepalived-ipfailover:latest] 437556283} {[openshift/origin-haproxy-router:de5b801 openshift/origin-haproxy-router:latest] 431811956} {[docker.io/openshift/origin-haproxy-router:latest] 431811956} {[openshift/origin:de5b801 openshift/origin:latest] 543747344} {[openshift/openvswitch:de5b801 openshift/openvswitch:latest] 228139043} {[openshift/origin-pod:de5b801 openshift/origin-pod:latest] 1590312} {[docker.io/openshift/origin-pod:latest] 1590312} {[rhel7.2:latest] 256294595} {[rhel7:latest rhel7.1:latest] 760499705} {[openshift/origin-release:latest] 711007836} {[openshift/origin-haproxy-router-base:latest] 298871240} {[openshift/origin-base:latest] 279586464} {[] 539322236} {[docker.io/centos/ruby-22-centos7:latest] 443849262} {[docker.io/openshift/ruby-20-centos7:latest] 443310209} {[<none>:<none>] 125026258} {[docker.io/centos:centos7] 196677849} {[registry.access.redhat.com/rhel7.2:latest] 203268847} {[gcr.io/google_containers/nettest:1.7] 24051275} {[gcr.io/google_containers/netexec:1.4] 7293444} {[gcr.io/google_containers/busybox:1.24] 1113554} {[gcr.io/google_containers/mounttest:0.6] 2084693} {[gcr.io/google_containers/hostexec:1.2] 13185747} {[gcr.io/google_containers/mounttest-user:0.3] 1718853} {[registry.access.redhat.com/rhel7.1:latest] 158229901} {[gcr.io/google_containers/mounttest:0.5] 1718853} {[gcr.io/google_containers/pause:2.0] 350164} {[gcr.io/google_containers/porter:cd5cb5791ebaa8641955f0e8c2a9bed669b1eaab] 5010921} {[gcr.io/google_containers/volume-iscsi:0.1] 244433376} {[docker.io/openshift/deployment-example:v1] 5769840} {[gcr.io/google_containers/serve_hostname:1.1] 4522409} {[gcr.io/google_containers/update-demo:nautilus] 4555533} {[gcr.io/google_containers/nginx:1.7.9] 91641000} {[gcr.io/google_containers/test-webserver:e2e] 4534272} {[gcr.io/google_containers/busybox:latest] 2429728} {[gcr.io/google_containers/liveness:e2e] 4387474}]}}
Apr 20 20:12:40.500: INFO: 
Logging kubelet events for node 172.18.14.36
Apr 20 20:12:40.502: INFO: 
Logging pods the kubelet thinks is on node 172.18.14.36
Apr 20 20:12:40.515: INFO: cephfs-server started at <nil> (0 container statuses recorded)
Apr 20 20:12:40.515: INFO: pi-4oaao started at <nil> (0 container statuses recorded)
Apr 20 20:12:40.515: INFO: router-2-zd6xz started at <nil> (0 container statuses recorded)
Apr 20 20:12:40.515: INFO: docker-registry-1-cdws8 started at <nil> (0 container statuses recorded)
Apr 20 20:12:40.515: INFO: update-demo-nautilus-yec2m started at <nil> (0 container statuses recorded)
Apr 20 20:12:40.992: INFO: ERROR kubelet_docker_errors{operation_type="inspect_container"} => 37 @[0]
Apr 20 20:12:40.992: INFO: ERROR kubelet_docker_errors{operation_type="inspect_image"} => 21 @[0]
Apr 20 20:12:40.992: INFO: ERROR kubelet_docker_errors{operation_type="start_container"} => 1 @[0]
Apr 20 20:12:40.992: INFO: ERROR kubelet_docker_errors{operation_type="stop_container"} => 79 @[0]
Apr 20 20:12:40.992: INFO: 
Latency metrics for node 172.18.14.36
Apr 20 20:12:40.992: INFO: {Operation: Method:pod_start_latency_microseconds Quantile:0.99 Latency:2m13.982561s}
Apr 20 20:12:40.992: INFO: {Operation: Method:pod_worker_start_latency_microseconds Quantile:0.99 Latency:1m33.730012s}
Apr 20 20:12:40.992: INFO: {Operation: Method:pod_start_latency_microseconds Quantile:0.9 Latency:37.919097s}
Apr 20 20:12:40.992: INFO: {Operation:update Method:pod_worker_latency_microseconds Quantile:0.99 Latency:31.994272s}
Apr 20 20:12:40.992: INFO: {Operation:update Method:pod_worker_latency_microseconds Quantile:0.9 Latency:30.787124s}
Apr 20 20:12:40.992: INFO: {Operation:stop_container Method:docker_operations_latency_microseconds Quantile:0.99 Latency:30.477712s}
Apr 20 20:12:40.992: INFO: {Operation:create Method:pod_worker_latency_microseconds Quantile:0.99 Latency:14.714274s}
Apr 20 20:12:40.992: INFO: {Operation:SyncPod Method:container_manager_latency_microseconds Quantile:0.99 Latency:12.837141s}
Apr 20 20:12:40.992: INFO: Unknown output type: . Skipping.
Apr 20 20:12:40.992: INFO: Waiting up to 1m0s for all nodes to be ready
STEP: Destroying namespace "extended-test-job-controller-uppzo-egdz2" for this suite.


• Failure [155.996 seconds]
[job] openshift can execute jobs
/data/src/github.com/openshift/origin/test/extended/jobs/jobs.go:49
  controller
  /data/src/github.com/openshift/origin/test/extended/jobs/jobs.go:48
    should create and run a job in user project [It]
    /data/src/github.com/openshift/origin/test/extended/jobs/jobs.go:47

    Expected error:
        <*errors.errorString | 0xc20802b6d0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    not to have occurred

    /data/src/github.com/openshift/origin/test/extended/jobs/jobs.go:30
------------------------------
```

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #8584.
@smarterclayton I've went with `gcr.io/google_containers/busybox` since we're already using it in networking tests, but that can be changed if you want me to. The current tests just `exit 0` and checks both `extensions/v1beta` and `batch/v1` jobs.

### CODE REVIEW

**Comment by soltysh** [comment]
[testonlyextended][extended:core(job)]

**Comment by soltysh** [comment]
AWS failure: `RequestLimitExceeded => Request limit exceeded. (Fog::Compute::AWS::Error)`.

[testonlyextended][extended:core(job)]

**Comment by soltysh** [comment]
Flake from https://github.com/openshift/origin/issues/7619

[testonlyextended][extended:core(job)]

**Comment by soltysh** [comment]
Jenkins why you don't like me today, it's such a beautiful day today :sob:

**Comment by soltysh** [comment]
@mfojtik this is your test, that's what git blame says :stuck_out_tongue_winking_eye: 
let me rephrase my focus:

[testonlyextended][extended:core(openshift can execute jobs)]

**Comment by soltysh** [comment]
Jenkins, now you're talking :-)

**Comment by smarterclayton** [comment]
Approved (test flake) [merge]

**Comment by smarterclayton** [comment]
[merge] on issue with setup versions

**Comment by smarterclayton** [comment]
[merge] import flake

**Comment by smarterclayton** [comment]
Saw https://ci.openshift.redhat.com/jenkins/view/Origin%20Test%20Jobs/job/origin_extended/lastCompletedBuild/testReport/(root)/Extended/_job__openshift_can_execute_jobs_controller_should_create_and_run_a_job_in_user_project/

**Comment by soltysh** [comment]
I think you're talking about https://ci.openshift.redhat.com/jenkins/view/Origin%20Test%20Jobs/job/origin_extended/831/testReport/%28root%29/Extended/_job__openshift_can_execute_jobs_controller_should_create_and_run_a_job_in_user_project/ @mfojtik pointed me to, above doesn't have a job failure. That error failed due to:
```
Error from server: error when creating "/data/src/github.com/openshift/origin/test/extended/fixtures/jobs/v1beta1.yaml": User "extended-test-job-controller-61ztx-f2vj7-user" cannot create extensions.jobs in project "extended-test-job-controller-61ztx-f2vj7"
```
Which suggests that for RoleBinding for that project wasn't setup correctly, imho.
